activerecord/lib/active_record.rb

```diff
@@ -197,6 +197,39 @@ def self.default_timezone=(default_timezone)
 
   self.default_timezone = :utc
 
+  # The action to take when database query produces warning.
+  # Must be one of :ignore, :log, :raise, :report, or a custom proc.
+  # The default is :ignore.
+  singleton_class.attr_reader :db_warnings_action
+
+  def self.db_warnings_action=(action)
+    @db_warnings_action =
+      case action
+      when :ignore
+        nil
+      when :log
+        ->(warning) do
+          warning_message = "[#{warning.class}] #{warning.message}"
+          warning_message += " (#{warning.code})" if warning.code
+          ActiveRecord::Base.logger.warn(warning_message)
```

Comment:
This will load `ActiveRecord::Base` when this config is set, and we don't want that. But it inside a `on_load` block.

Reply:
Ah, right.